test/pkg/environment/common/expectations.go

```diff
@@ -927,13 +929,9 @@ func (env *Environment) GetDaemonSetOverhead(np *corev1beta1.NodePool) v1.Resour
 	})...)
 }
 
-func (env *Environment) EventuallyExpectNodeClassStatusCondition(nodeClass *v1beta1.EC2NodeClass, condition string, status bool, message string) {
+func (env *Environment) EventuallyExpectStatusCondition(nodeClass *v1beta1.EC2NodeClass, conditions ...status.Condition) {
 	GinkgoHelper()
 	Eventually(func(g Gomega) {
-		nc := &v1beta1.EC2NodeClass{}
-		g.Expect(env.Client.Get(env, client.ObjectKeyFromObject(nodeClass), nc)).To(Succeed())
-		statusCondition := nc.StatusConditions().Get(condition)
-		g.Expect(statusCondition.IsTrue()).To(Equal(status))
-		g.Expect(statusCondition.Message).To(Equal(message))
+		op.ExpectStatusConditions(nodeClass, conditions...)
```

Comment:
This expectation doesn't work with the `g` context, so it won't keep retrying. It if ever fails it'll just bail out immediately